controllers/topics/index.js

```diff
@@ -189,17 +218,30 @@ const followTopicV2 = async (req, res) => {
   }
 };
 
-const _afterPutTopic = async (topic, token, userId, name) => {
+const _afterPutTopic = async (isUnfollow, token, userId, name) => {
   // follow / unfollow main feed topic
-  if (topic) {
+  if (isUnfollow) {
     await unfollowMainFeedTopic(token, userId, name);
   } else {
     await followMainFeedTopic(token, userId, name);
   }
 
-  const message = topic ? 'Success delete topic user v2' : 'Success add topic user v2';
+  let data;
+  if (isUnfollow) {
+    data = {
+      message: 'Success delete topic user v2',
+      type: 'unfollow',
+      userId
+    };
+  } else {
+    data = {
+      message: 'Success add topic user v2',
+      type: 'follow',
+      userId
+    };
+  }
 
-  return message;
+  return data;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `_afterPutTopic` function now returns an object instead of a string, which is a good change for better structured data. However, the `data` object creation is duplicated in the if-else block. You can simplify this by defining the common properties first and then adding the conditional property.

```diff
  let data = {
    userId,
    type: isUnfollow ? 'unfollow' : 'follow'
  };
  
  if (isUnfollow) {
    data.message = 'Success delete topic user v2';
  } else {
    data.message = 'Success add topic user v2';
  }
 
  return data;
```
This way, you avoid duplicating the structure of the `data` object and make your code more DRY (Don't Repeat Yourself).


<!-- This is an auto-generated comment by OSS CodeRabbit -->